### DISCUSSION TITLE
Is it possible to subclass the rootwidget?

### DISCUSSION THREAD

**Discussion by BurningTreeC** [TiddlyWiki/TiddlyWiki5#6629] [Q&A]
Hi all,

I have a question regarding the `rootwidget`
I'd like to extend a method in a plugin I'm writing
Is that possible through something like a subclassing mechanism?

Best wishes,
Simon

**Comment by Jermolene** [comment]
Hi @BurningTreeC if you only want your new properties to be available on the root widget, and not other instances of the widget base class, then you should be able to poke those methods into the existing instance with something like `$tw.rootWidget.myMethod = function() {...}`.

**Reply by BurningTreeC** [reply]
Thank you @Jermolene , I think I haven't been precise enough
In the exports.startup function I'd like to modify an EventListener without overwriting the core rootwidget.js tiddler
Is that even possible?

**Reply by Jermolene** [reply]
It sounds like those event listeners would need to be refactored as methods that you could override. What is the tm- message you are thinking about?

**Reply by BurningTreeC** [reply]
Thanks for your answer @Jermolene . The message I'd like to override is the `tm-focus-selector` message. I really want to apply an optional delay to it through `setTimeout`

**Reply by Jermolene** [reply]
Perhaps you could implement your variant as a separate message `tm-focus-selector-deferred`?
